rex/rechunk_h5/chunk_size.py

```diff
@@ -0,0 +1,259 @@
+        -------
+        chunks : tuple
+            Dataset chunk size along all axis
+        """
+        time_chunk = cls._compute_time_chunk_size(
+            shape[0],
+            weeks_per_chunk=weeks_per_chunk)
+        site_chunk = cls._compute_site_chunk_size(time_chunk, dtype,
+                                                  chunk_size=chunk_size)
+
+        sites = np.sum(shape[1:])
+        if site_chunk >= sites:
+            site_chunk = shape[1:]
+        elif len(shape) > 2:
+            weights = [i // sites for i in shape[1:]]
+            site_chunk = (int(w * site_chunk) for w in weights)
+        else:
+            site_chunk = (site_chunk, )
+
+        chunks = (time_chunk, ) + site_chunk
+
+        return chunks
+
+    @classmethod
+    def compute(cls, shape, dtype, chunk_size=2, weeks_per_chunk=None):
+        """
+        Compute chunks based on array shape and dtype
+
+        Parameters
+        ----------
+        shape : tuple
+            Array shape
+        dtype : str | np.dtype
+            Array data type
+        chunk_size : int, optional
+            Chunk size in MB, by default 2
+        weeks_per_chunk : int, optional
+            Number of weeks per time chunk, if None scale weeks based on 8
+            weeks for hourly data, by default None
+
+        Returns
+        -------
+        chunks : tuple
+            Dataset chunk size along all axis
+        """
+        chunks = cls(shape, dtype, chunk_size=chunk_size,
+                     weeks_per_chunk=weeks_per_chunk)
+
+        return chunks.chunks
+
+
+class ArrayChunkSize(BaseChunkSize):
+    """
+    Compute chunks based on array size
+    """
+    def __init__(self, arr, chunk_size=2):
+        """
+        Parameters
+        ----------
+        arr : ndarray
+            Dataset array
+        chunk_size : int, optional
+            Chunk size in MB, by default 2
+        """
+        self._chunks = self.compute_arr_chunks(arr, chunk_size=chunk_size)
+
+    @staticmethod
+    def _get_arr_size(arr):
+        """
+        Get the size of the given array in MB
+
+        Parameters
+        ----------
+        data : ndarray
+            Data array to chunk
+
+        Returns
+        -------
+        float
+            Data array size in MB
+        """
+
+        return arr.size * arr.dtype.itemsize * 10**-6
+
+    @classmethod
+    def compute_arr_chunks(cls, arr, chunk_size=2):
+        """
+        Compute chunks based on array size
+
+        Parameters
+        ----------
+        arr : ndarray
+            Dataset array
+        chunk_size : int, optional
+            Chunk size in MB, by default 2
+
+        Returns
+        -------
+        chunks : tuple
+            Dataset chunk size along all axis
+        """
+        arr_size = cls._get_arr_size(arr)
+        if arr_size >= 2:
+            arr_len = len(arr)
+            chunks = int(arr_len // (arr_size / chunk_size))
+            chunks = (chunks, ) + arr.shape[1:]
```

Comment:
Is this only computing chunks in the row direction? Should specify that in the doc strings. Also, as above, i don't think we should use `int()` for the very minor risk that it rounds to zero.

Reply:
will do